src/inmanta/agent/agent_new.py

```diff
@@ -206,51 +208,65 @@ async def start_working(self) -> None:
         await self.executor_manager.start()
         await self.scheduler.start()
         self._enable_time_triggers()
+        LOGGER.info("Scheduler started for environment %s", self.environment)
 
-    async def stop_working(self) -> None:
+    async def stop_working(self, timeout: float = 0.0) -> None:
         """Stop working, connection lost"""
         if not self.working:
             return
         self.working = False
         self._disable_time_triggers()
-        await self.executor_manager.stop()
         await self.scheduler.stop()
+        await self.executor_manager.stop()
+        await self.executor_manager.join([], timeout=timeout)
+        await self.scheduler.join()
+        LOGGER.info("Scheduler stopped for environment %s", self.environment)
 
     @protocol.handle(methods_v2.update_agent_map)
     async def update_agent_map(self, agent_map: dict[str, str]) -> None:
         # Not used here
         pass
 
-    async def unpause(self, name: str) -> Apireturn:
-        if name != AGENT_SCHEDULER_ID:
-            return 404, "No such agent"
-
-        LOGGER.info("Scheduler started for environment %s", self.environment)
-        await self.start_working()
-        return 200
-
-    async def pause(self, name: str) -> Apireturn:
-        if name != AGENT_SCHEDULER_ID:
-            return 404, "No such agent"
-
-        LOGGER.info("Scheduler stopped for environment %s", self.environment)
-        await self.stop_working()
-        return 200
-
     @protocol.handle(methods.set_state)
     async def set_state(self, agent: str, enabled: bool) -> Apireturn:
-        if enabled:
-            return await self.unpause(agent)
+        if agent == AGENT_SCHEDULER_ID:
+            should_be_running = await self.scheduler.should_be_running()
```

Comment:
```suggestion
            should_be_running = enabled
```
it becomes too complicated otherwise